src/apps/DesktopAppPage.vala

```diff
@@ -400,5 +563,470 @@ namespace Ilia {
+                            hide_app_actions();
+                            session_controller.quit();
+                        } catch (Error e) {
+                            stderr.printf("Failed to open app folder: %s\n", e.message);
+                            hide_app_actions();
+                        }
+                        break;
+                        
+                    case "generic-new-workspace":
+                        try {
+                            // Get the app command
+                            var key_file = new KeyFile();
+                            key_file.load_from_file(desktop_file_path, KeyFileFlags.NONE);
+                            string exec_line = key_file.get_string("Desktop Entry", "Exec");
+                            string clean_exec = "";
+                            bool in_arg = false;
+                            for (int i = 0; i < exec_line.length; i++) {
+                                if (exec_line[i] == '%') {
+                                    in_arg = true;
+                                    continue;
+                                }
+                                if (in_arg) {
+                                    in_arg = false;
+                                    continue;
+                                }
+                                clean_exec += exec_line[i].to_string();
+                            }
+                            clean_exec = clean_exec.strip();
+                            string[] args;
+                            var wm_cmd_newws = get_wm_cli(this.session_controller.get_wm_name());
+                            if (wm_cmd_newws != null) {
+                                // For i3/sway: switch to next workspace and launch app
+                                string launch_cmd = wm_cmd_newws + " workspace next && " + clean_exec;
+                                args = {"sh", "-c", launch_cmd};
+                            } else {
+                                args = {"sh", "-c", clean_exec};
+                            }
+                            Process.spawn_async(null, args, null, SpawnFlags.SEARCH_PATH, null, null);
+                            hide_app_actions();
+                            session_controller.quit();
+                        } catch (Error e) {
+                            stderr.printf("Failed to launch in new workspace: %s\n", e.message);
+                            hide_app_actions();
+                        }
+                        return;
+                        
+                    default:
+                        hide_app_actions();
+                        break;
+                }
+                return;
+            }
+            
+            // Launch regular app action
+            try {
+                // Create empty AppLaunchContext instead of passing null
+                AppLaunchContext context = new AppLaunchContext();
+                app_info.launch_action(action_id, context);
```

Comment:
Check if it is possible to make this use `systemd-run` as well. Anything that ilia tries to execute should ideally use that command.